### PR TITLE
Remove Yahoo API, introduce guzzle

### PR DESCRIPTION
**Pull request by jbrinksmeier**

### CODE REVIEW

File: src/Provider/FixerApi.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        $path = sprintf(
+            self::FIXER_API_BASEPATH . '?symbols=%s&base=%s',
+            $toCurrency,
+            $fromCurrency
+        );
+        $result = json_decode($this->httpClient->get($path)->getBody(), true);
+        return $result['rates'][$toCurrency];
+    }
+}
```
Comment (ojhaujjwal) [inline src/Provider/FixerApi.php]: new line please at the end of every file.

File: src/Provider/YahooApi.php [hunk 2]
```diff
@@ -1,38 +0,0 @@
...
-    public function getRate($fromCurrency, $toCurrency)
-    {
-        $fromCurrency = urlencode($fromCurrency);
-        $toCurrency = urlencode($toCurrency);
-
-        $url = str_replace(
-            ['[fromCurrency]', '[toCurrency]'],
-            [$fromCurrency, $toCurrency],
-            static::API_URL
-        );
```
Comment (ojhaujjwal) [inline src/Provider/YahooApi.php]: `sprintf` is probably more performant than str_replace especially in predictable string replace scenarios. I don't have the stats. But, that's my guess.

File: src/Provider/FixerApi.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    protected $httpClient;
+
+    /**
+     * FixerApi constructor.
+     * @param Client $httpClient
+     */
+    public function __construct(Client $httpClient=null)
+    {
+        $this->httpClient = $httpClient;
```
Comment (ojhaujjwal) [inline src/Provider/FixerApi.php]: please, create a default client here as well.

**Comment by jbrinksmeier** [comment]
there you are

**Comment by ojhaujjwal** [comment]
thanks for the contribution man. Cheers!

**Comment by jbrinksmeier** [comment]
my pleasure :)

**Pull request merged**